### DISCUSSION TITLE
[Why?] MagicReadSettings setup in MagickImage

### DISCUSSION THREAD

**Discussion by Gounlaf** [dlemstra/Magick.NET#1495] [General] [answered]
Good evening,

I've question regarding the readSettings setup in Read or Ping functions:
- sometimes the settings are created from existing ones : https://github.com/dlemstra/Magick.NET/blob/main/src/Magick.NET/MagickImage.cs#L4904
- sometimes it doesn't (when MagickFormat is provided) ; https://github.com/dlemstra/Magick.NET/blob/main/src/Magick.NET/MagickImage.cs#L4753

What the reason behind?

Thanks.

Regards.

**Comment by dlemstra** [comment, marked as answer]
I think you found a mistake in the code. The `_settings` should be used when creating the new `MagicReadSettings`. Thanks for pointing this out. I just pushed a patch to fix this.

**Reply by Gounlaf** [reply]
> Thanks for pointing this out

:handshake: :wink:
